python/cugraph/cugraph/structure/graph_classes.py

```diff
@@ -17,6 +17,7 @@
                                    npartiteGraphImpl)
 import cudf
 import warnings
+import pandas
```

Comment:
This is a new dependency, so our conda recipe has to be updated.  Maybe we should look into a different way to do the type checking that doesn't require depending on Pandas?

This would be a good candidate for an optional import, since users passing Pandas types should already have Pandas installed, and those without Pandas probably wouldn't pass those types.  [Here's](https://github.com/rapidsai/cugraph/blob/df2d81dbb26904775386a6a25f20b776cd1897cb/python/cugraph/cugraph/utilities/utils.py#L311) the utility, and [here](https://github.com/rapidsai/cugraph/blob/df2d81dbb26904775386a6a25f20b776cd1897cb/python/cugraph/cugraph/utilities/nx_factory.py#L19) and [here](https://github.com/rapidsai/cugraph/blob/df2d81dbb26904775386a6a25f20b776cd1897cb/python/cugraph/cugraph/utilities/nx_factory.py#L28) are examples.